src/mixins/java/org/spongepowered/common/mixin/core/world/entity/player/PlayerMixin.java

```diff
@@ -623,4 +625,18 @@ public void attack(final Entity targetEntity) {
     public boolean impl$canCallIgniteEntityEvent() {
         return !this.shadow$isSpectator() && !this.shadow$isCreative();
     }
+
+    @Inject(method = "canHarmPlayer", at = @At("HEAD"), cancellable = true)
+    private void impl$onCanHarmPlayer(final net.minecraft.world.entity.player.Player other, final CallbackInfoReturnable<Boolean> cir) {
+        if (!(other instanceof org.spongepowered.api.entity.living.player.server.ServerPlayer)) {
```

Comment:
`canHarmPlayer` also exists on `ServerPlayer` - why did you not inject at an appropriate point there instead, rather than do a type check here?

Reply:
To note, this is the paramater of the method, not `this`. I had it previously in the `ServerPlayer` but I decided to change its location.

Whenever you attack another player, you are the attacker (`this`) and the attacked is the `other`. So if the attacker isn't instance of ServerPlayer but the attacked is, then it wouldn't properly check against it.

Reply:
Yes, `other` - skipped over that detail, got it.